src/hotspot/cpu/riscv/sharedRuntime_riscv.cpp

```diff
@@ -190,10 +190,10 @@ OopMap* RegisterSaver::save_live_registers(MacroAssembler* masm, int additional_
     oop_map->set_callee_saved(VMRegImpl::stack2reg(sp_offset_in_slots), r->as_VMReg());
   }
 
-  // ignore zr, ra and sp, being ignored also by push_CPU_state (pushing zr only for stack alignment)
+  // ignore zr, ra, sp, gp and tp, being ignored also by push_CPU_state (pushing zr only for stack alignment)
   sp_offset_in_slots += RegisterImpl::max_slots_per_register;
   step_in_slots = RegisterImpl::max_slots_per_register;
-  for (int i = 3; i < RegisterImpl::number_of_registers; i++, sp_offset_in_slots += step_in_slots) {
+  for (int i = 5; i < RegisterImpl::number_of_registers; i++, sp_offset_in_slots += step_in_slots) {
     Register r = as_Register(i);
     if (r != xthread && r != t0 && r != t1) {
```

Comment:
Why not start the loop from x7?

Reply:
Thanks for the nice reminder -- changed.

Reply:
Need some testing work for the new change -- I will ping this thread after tests are done.

Reply:
`test/jtreg/hotspot/compiler` on Qemu reveals no other errors.